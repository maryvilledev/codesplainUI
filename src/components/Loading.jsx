import React from 'react';
import { CircularProgress } from 'material-ui'

const styles = {
  centered: {
    color: '#a6a6a6',
    fontSize: '15pt',
    position: 'fixed',
    top: '50%',
    left: '50%',
    marginTop: '-100px',
    marginLeft: '-50px',
  },
}

/*
<Loading /> renders as a <CircularProgress /> in the center of the screen with
the text passed in with the 'text' prop displayed above it.
*/
const Loading = ({ text }) => (
  <div style={styles.centered}>
    <p>{text}</p>
    <CircularProgress 
      size={100} 
      thickness={7} 
    />
  </div>
);

export default Loading