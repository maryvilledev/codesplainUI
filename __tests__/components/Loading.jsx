import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import Loading from '../../src/components/Loading';

describe('<Loading />', () => {
  it('matches the stored snapshot', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <Loading />
    );
    expect(tree).toMatchSnapshot();
  });
});
