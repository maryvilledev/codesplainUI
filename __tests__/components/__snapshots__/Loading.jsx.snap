exports[`<Loading /> matches the stored snapshot 1`] = `
<div
  style={
    Object {
      "color": "#a6a6a6",
      "fontSize": "15pt",
      "left": "50%",
      "marginLeft": "-50px",
      "marginTop": "-100px",
      "position": "fixed",
      "top": "50%",
    }
  }>
  <p />
  <CircularProgress
    max={100}
    min={0}
    mode="indeterminate"
    size={100}
    thickness={7}
    value={0} />
</div>
`;
